### ISSUE TITLE
Direct access to logits (add `predict_logits` to `TabPFNClassifier`)

### ISSUE THREAD

**Issue by mmschlk** [PriorLabs/TabPFN#374]
### Describe the workflow you want to enable

We would like to enable the **following workflow**:

For a classification task, I would like to make predictions with `TabPFNClassifier` and retrieve the **raw logits** (before they go through the softmax).

### Describe your proposed solution

We would either:
- **(a)** add a `predict_logits` function to `TabPFNClassifier` which works similarly to `predict_proba` but returns the raw logits instead of the probabilities (this function could be even called inside `predict_proba` to not violate DRY), or
- **(b)** add a `logits: bool = False` flag to `TabPFNClassifier.predict_proba` which if set to `True` would return the logits instead of the probabilities.

We see that you are doing many complicated things with the different `_OUTPUT_TYPES` variants in `TabPFNRegressor`. We don't know if it is necessary to map this logic from `TabPFNRegressor` also into `TabPFNClassifier`, as this would probably be a rather big refactoring. 

### Describe alternatives you've considered, if relevant

Currently, we are using a nasty wrapper around tabpfn (substitute `pfn` with `self` and it works as an instance function of `TabPFNClassifier`):

```python
def tabpfn_predict_logits(pfn, X):
    X = validate_X_predict(X, pfn)
    X = _fix_dtypes(X, cat_indices=pfn.categorical_features_indices)

    X = _process_text_na_dataframe(X, ord_encoder=pfn.preprocessor_)

    outputs: list[torch.Tensor] = []

    for output, config in pfn.executor_.iter_outputs(
        X,
        device=pfn.device_,
        autocast=pfn.use_autocast_,
    ):
        assert isinstance(config, ClassifierEnsembleConfig)
        # Cut out logits for classes which do not exist
        assert output.ndim == 2

        if pfn.softmax_temperature != 1:
            output = (  # noqa: PLW2901
                output[:, : pfn.n_classes_].float() / pfn.softmax_temperature
            )

        # Reverse class permutation if exists
        if config.class_permutation is not None:
            output = output[..., config.class_permutation]  # noqa: PLW2901

        outputs.append(output)


    output = torch.stack(outputs).mean(dim=0)


    output = output.float().cpu().numpy()


    return output
```

**Note:** This wrapper does not care about the `USE_SKLEARN_16_DECIMAL_PRECISION` yet, which would need to be addressed.

### Additional context

Hi all,

@pwhofman and me are working on **uncertainty quantification** and **XAI**. We are heavily using tabpfn for this. To do this right, we would really need access to the logits of the models instead of the probabilities when doing predictions.

For example, if you do an explanation with Shapley values (i.e. with `shapiq.TabPFNExplainer` 😉) then you would like the output you are explaining to be on an additive scale. This is naturally true for regression models because then your model map from $f \rightarrow \mathbb{R}$. For classification this is only true if we get the logits otherwise the models map into a limited space $f \rightarrow [0, 1]$. While explaining these probas still works, it is much more natural to explain something mapping into  $\mathbb{R}$.

**We would love to add this as a pull request**, but did not know what your philosophy on adding features is and wanted to first know what you think of this. Also, we were not sure if we maybe missed something super obvious until now.

Best
Max and Paul

PS.: ❤️ your model!

### Impact

High (Major improvement)

**Comment by noahho** [comment]
Hope our PR solved this!

**Comment by mmschlk** [comment]
Amazing! Will give it a shot and adjust the `shapiq.TabPFNExplainer` exposed in the `tabpfn_extensions`.

**Comment by noahho** [comment]
Nice! Sorry I didn't communciate properly that we did this three weeks ago! Let me know how things go :-)
